great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -547,12 +548,26 @@ def _estimate_metric_value_range(
                 metric_value_idx
             ] = numeric_range_estimation_result.estimation_histogram
             # Store computed min and max value estimates into allocated range estimate for multi-dimensional metric.
-            metric_value_range[metric_value_range_min_idx] = round(
-                cast(float, min_value), round_decimals
-            )
-            metric_value_range[metric_value_range_max_idx] = round(
-                cast(float, max_value), round_decimals
-            )
+            if len(metric_values) == 1 and round_decimals != 0:
+                # If a range wasn't created due to only having one batch,
+                # create a range with precision less than the metric value
+                metric_precision: float = float(str(metric_values[0])[::-1].find("."))
```

Comment:
@NathanFarmer This does not feel right -- I actually suspect that `Expectation` is being too strict instead of doing the "near" comparison.